src/hooks/screen/useCreatePostHook.js

```diff
@@ -12,27 +12,22 @@ import AnonUserInfoRepo from '../../service/repo/anonUserInfoRepo';
 const useCreatePostHook = (isAnonymous) => {
   const {params = {}} = useRoute();
   const {topic} = params;
+  const [selectedTopic, setSelectedTopic] = React.useState(topic);
   const [anonUserInfo, setAnonUserInfo] = React.useState(null);
+  const [headerTitle, setHeaderTitle] = React.useState(
+    selectedTopic ? `Create Post in #${selectedTopic}` : 'Create Post'
+  );
 
-  const headerTitle = topic ? `Create Post in #${topic}` : 'Create Post';
-
-  const isInCreatePostTopicScreen = !!topic;
+  const isInCreatePostTopicScreen = !!selectedTopic;
 
   const getAnonUserInfo = React.useCallback(async () => {
     try {
       const response = await AnonUserInfoRepo.getPostAnonUserInfo();
-      if (response.isSuccess) {
-        setAnonUserInfo(response.data);
-      } else {
-        showMessage({
-          message: response.error,
-          type: 'danger'
-        });
-      }
+      setAnonUserInfo(response.data);
     } catch (e) {
       console.log('Error in getAnonUserInfo', e);
       showMessage({
-        message: e,
+        message: JSON.stringify(e),
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling here could be improved. Currently, the error message is simply stringified and displayed to the user, which might not be very helpful or user-friendly. Consider providing a more descriptive error message.

```diff
- showMessage({
-  message: JSON.stringify(e),
-  type: 'danger'
- });
+ showMessage({
+  message: "An error occurred while fetching anonymous user info.",
+  type: 'danger'
+ });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->